Incorrect invocation of ScalaStyleUtils.getTestFilesToProcess() in ScalaStyleTask.groovy

There appears to be an issue with attempting to use a separate test config.  The plugin attempts to invoke <code>ScalaStyleUtils.getTestFilesToProcess()</code> with incorrect arguments on line 71 of <code>ScalaStyleTask.groovy</code>.

gradle.build scalastyle configuration:

<pre>
scalaStyle {
    configLocation = "gradle/plugin/config/scalastyle/scalastyle_src_config.xml"
    testConfigLocation = "gradle/plugin/config/scalastyle/scalastyle_spec_config.xml"
    includeTestSourceDirectory = false
    source = "src/main/scala"
    testSource = "src/test/scala"
    verbose = true
}
</pre>


Stacktrace:

<pre>
mcosxh01:algorithm5 mcibulskis$ ./gradlew --stacktrace scalastyle
:scalaStyle FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':scalaStyle'.
> java.lang.Exception: Scala check error

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':scalaStyle'.
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
    at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
    at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
    at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
    at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
    at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
    at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
    at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
    at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
    at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
    at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
    at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
    at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:25)
    at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:110)
    at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
    at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
    at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
    at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
    at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
    at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
    at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
    at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:154)
    at org.gradle.internal.Factories$1.create(Factories.java:22)
    at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
    at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:52)
    at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:151)
    at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:32)
    at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:99)
    at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:93)
    at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
    at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:62)
    at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:93)
    at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:82)
    at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:94)
    at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
    at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
    at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:43)
    at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
    at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:75)
    at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:45)
    at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:51)
    at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:28)
    at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:43)
    at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:170)
    at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:237)
    at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:210)
    at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:35)
    at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
    at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:206)
    at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
    at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
    at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
    at org.gradle.launcher.Main.doAction(Main.java:33)
    at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
    at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:54)
    at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:35)
    at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
    at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:30)
    at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:129)
    at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
Caused by: org.gradle.internal.UncheckedException: java.lang.Exception: Scala check error
    at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:45)
    at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:78)
    at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:227)
    at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:220)
    at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:209)
    at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:585)
    at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:568)
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
    at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
    ... 60 more
Caused by: java.lang.Exception: Scala check error
    at org.github.ngbinh.scalastyle.ScalaStyleTask.scalaStyle(ScalaStyleTask.groovy:95)
    at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
    ... 67 more
Caused by: groovy.lang.MissingMethodException: No signature of method: org.github.ngbinh.scalastyle.ScalaStyleUtils.getTestFilesToProcess() is applicable for argument types: (java.util.ArrayList, java.util.ArrayList, java.lang.String, java.lang.Boolean) values: [[/Volumes/projects-personal/algorithm5/src/main/scala/com/noackexpected/algorithm5/cluster/InMemoryNeighborInformation.scala, ...], ...]
Possible solutions: getTestFilesToProcess(java.util.List, java.lang.String, boolean)
    at org.github.ngbinh.scalastyle.ScalaStyleTask.scalaStyle(ScalaStyleTask.groovy:71)
    ... 68 more


BUILD FAILED

Total time: 3.381 secs
</pre>

--- a/gradle-scalastyle-plugin_2.10/src/main/groovy/org/github/ngbinh/scalastyle/ScalaStyleTask.groovy
+++ b/gradle-scalastyle-plugin_2.10/src/main/groovy/org/github/ngbinh/scalastyle/ScalaStyleTask.groovy
@@ -20,13 +20,9 @@ package org.github.ngbinh.scalastyle
 import org.gradle.api.file.FileTree
 import org.gradle.api.tasks.SourceTask
 import org.gradle.api.tasks.TaskAction
-import org.scalastyle.ScalastyleChecker
 import org.scalastyle.ScalastyleConfiguration
 import org.scalastyle.TextOutput
 import org.scalastyle.XmlOutput
-import com.typesafe.config.ConfigFactory
-import com.typesafe.config.Config
-
 /**
  * @author Binh Nguyen
  * @since 12/16/2014
@@ -68,7 +64,7 @@ class ScalaStyleTask extends SourceTask {
                 if (testConfigLocation != null) {
                   def testConfiguration = ScalastyleConfiguration.readFromXml(testConfigLocation)
                   if (testConfiguration != null) {
-                      def testFilesToProcess = scalaStyleUtils.getTestFilesToProcess(source.getFiles().toList(), testSourceDir.getFiles().toList(), inputEncoding, includeTestSourceDirectory)
+                      def testFilesToProcess = scalaStyleUtils.getTestFilesToProcess(testSourceDir.getFiles().toList(), inputEncoding)
                       messages.addAll(scalaStyleUtils.checkFiles(testConfiguration, testFilesToProcess))
                   }
                 }
@@ -120,7 +116,7 @@ class ScalaStyleTask extends SourceTask {
             throw new Exception("Specify Scala source set")
         }
 
-        if (includeTestSourceDirectory && testSource == null) {
+        if (testSource == null) {
             testSourceDir = project.fileTree(project.projectDir.absolutePath + "/src/test/scala")
         } else {
             testSourceDir = project.fileTree(project.projectDir.absolutePath + "/" + testSource)

--- a/gradle-scalastyle-plugin_2.10/src/main/scala/org/github/ngbinh/scalastyle/ScalaStyleUtils.scala
+++ b/gradle-scalastyle-plugin_2.10/src/main/scala/org/github/ngbinh/scalastyle/ScalaStyleUtils.scala
@@ -41,7 +41,7 @@ class ScalaStyleUtils {
     sd ::: tsd
   }
 
-  def getTestFilesToProcess(testFiles: jList[File], inputEncoding: String, includeTestSourceDirectory: Boolean): List[FileSpec] = {
+  def getTestFilesToProcess(testFiles: jList[File], inputEncoding: String): List[FileSpec] = {
     getFiles("testFiles", asScalaBufferConverter(testFiles).asScala.toList, inputEncoding)
   }
 

--- a/gradle-scalastyle-plugin_2.11/src/main/groovy/org/github/ngbinh/scalastyle/ScalaStyleTask.groovy
+++ b/gradle-scalastyle-plugin_2.11/src/main/groovy/org/github/ngbinh/scalastyle/ScalaStyleTask.groovy
@@ -20,13 +20,9 @@ package org.github.ngbinh.scalastyle
 import org.gradle.api.file.FileTree
 import org.gradle.api.tasks.SourceTask
 import org.gradle.api.tasks.TaskAction
-import org.scalastyle.ScalastyleChecker
 import org.scalastyle.ScalastyleConfiguration
 import org.scalastyle.TextOutput
 import org.scalastyle.XmlOutput
-import com.typesafe.config.ConfigFactory
-import com.typesafe.config.Config
-
 /**
  * @author Binh Nguyen
  * @since 12/16/2014
@@ -68,7 +64,7 @@ class ScalaStyleTask extends SourceTask {
                 if (testConfigLocation != null) {
                   def testConfiguration = ScalastyleConfiguration.readFromXml(testConfigLocation)
                   if (testConfiguration != null) {
-                      def testFilesToProcess = scalaStyleUtils.getTestFilesToProcess(source.getFiles().toList(), testSourceDir.getFiles().toList(), inputEncoding, includeTestSourceDirectory)
+                      def testFilesToProcess = scalaStyleUtils.getTestFilesToProcess(testSourceDir.getFiles().toList(), inputEncoding)
                       messages.addAll(scalaStyleUtils.checkFiles(testConfiguration, testFilesToProcess))
                   }
                 }
@@ -120,7 +116,7 @@ class ScalaStyleTask extends SourceTask {
             throw new Exception("Specify Scala source set")
         }
 
-        if (includeTestSourceDirectory && testSource == null) {
+        if (testSource == null) {
             testSourceDir = project.fileTree(project.projectDir.absolutePath + "/src/test/scala")
         } else {
             testSourceDir = project.fileTree(project.projectDir.absolutePath + "/" + testSource)

--- a/gradle-scalastyle-plugin_2.11/src/main/scala/org/github/ngbinh/scalastyle/ScalaStyleUtils.scala
+++ b/gradle-scalastyle-plugin_2.11/src/main/scala/org/github/ngbinh/scalastyle/ScalaStyleUtils.scala
@@ -41,7 +41,7 @@ class ScalaStyleUtils {
     sd ::: tsd
   }
 
-  def getTestFilesToProcess(testFiles: jList[File], inputEncoding: String, includeTestSourceDirectory: Boolean): List[FileSpec] = {
+  def getTestFilesToProcess(testFiles: jList[File], inputEncoding: String): List[FileSpec] = {
     getFiles("testFiles", asScalaBufferConverter(testFiles).asScala.toList, inputEncoding)
   }
 

